gibbon-compiler/src/Gibbon/L2/Syntax.hs

```diff
@@ -324,6 +351,7 @@ data Region = GlobR Var Multiplicity -- ^ A global region with lifetime equal to
             | MMapR Var              -- ^ A region that doesn't result in an (explicit)
                                      --   memory allocation. It merely ensures that there
                                      --   are no free locations in the program.
+            | AnalyzedRegion Region RegionSize
```

Comment:
Instead of having a separate region variant here, what do you think about this:

```
data Region = {same as now}
data RegionSize = BoundedSize Int | Unbounded | Undefined
-- Since RegionSize already covers the "unbounded" or "unknown" sizes
data E2Ext = LetRegion Region RegionSize
           | {same as now}
```

OR

```
data Region = {same as now}
type RegionSize = Int
data E2Ext = LetRegion Region (Maybe RegionSize) Exp
           | {same as now}
```

OR if you don't want to change the `letregion` expression and tie a region with its bound like you have:

```
data Region = MkRegion RegionVariant RegionSize
data RegionVariant = GlobR ... | MMapR ... | ...
data RegionSize = BoundedSize Int | Unbounded | Undefined
```

Did you try something like this and run into trouble somewhere later?

Reply:
changing E2Ext would have required a lot of refactoring (around ~130 locations), lots of pattern matches and other things are tightly coupled with the data constructor, unless we had record fields, then it might have been ok.

Reply:
Making this type recursive still feels strange to me, but it's fine for now. In the future, this could probably be refactored as:
```
data RegionScope = Global | Local
data Region = VarR Var | MMapR Var | AnalyzedR Var (Maybe Int) RegionScope Multiplicity
```